app/services/record_declaration.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+class RecordDeclaration
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :course_identifier
+  attribute :cpd_lead_provider
+  attribute :declaration_date, :datetime
+  attribute :declaration_type
+  attribute :participant_id
+  attribute :evidence_held
+
+  before_validation :declaration_attempt
+
+  validates :declaration_date, :declaration_type, presence: true
+  validates :declaration_date, future_date: true, declaration_date: true, allow_blank: true
+  validates :participant_id, participant_not_withdrawn: true
+  validates :cpd_lead_provider, induction_record: true
+  validates :course_identifier, course: true
+  validates :evidence_held,
+            presence: { message: I18n.t(:missing_evidence_held), if: :validate_evidence_held? },
+            inclusion: {
+              in: ->(record_declaration) { record_declaration.participant_profile.class::VALID_EVIDENCE_HELD },
+              if: :validate_evidence_held?,
+              message: I18n.t(:invalid_evidence_type),
+            }
+
+  validate :validate_milestone_exists
+
+  attr_reader :raw_declaration_date
+
+  DECLARATION_TYPE_WITH_EVIDENCE_HELD = %w[
+    completed
+    retained-1
+    retained-2
+    retained-3
+    retained-4
+  ].freeze
+
+  def call
+    return if invalid?
+
+    ParticipantDeclaration.transaction do
+      set_eligibility
+
+      Finance::Statement.attach(participant_declaration) unless participant_declaration.submitted?
+
+      declaration_attempt.update!(participant_declaration:)
+    end
+
+    participant_declaration
+  end
+
+  def declaration_date=(raw_declaration_date)
+    self.raw_declaration_date = raw_declaration_date
+    super
+  end
+
+  def participant_identity
+    @participant_identity ||= ParticipantIdentity.find_by(external_identifier: participant_id)
+  end
+
+  def participant_profile
+    @participant_profile ||= ParticipantProfileResolver.call(participant_identity, course_identifier)
+  end
+
+  def milestone
+    return unless schedule
+
+    schedule.milestones.find_by(declaration_type:)
+  end
+
+  def schedule
+    participant_profile&.schedule
+  end
+
+private
```

Comment:
https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/services/record_declarations/base.rb#L94

I think the code has diverged and missing the above validation

Reply:
well spotted! re-added